quizmaster-client-app/src/components/QuizHosting/QuestionCreator/QuestionReducer.ts

```diff
@@ -0,0 +1,30 @@
+import Row from "./Row";
+
+function Reducer(state: any, action: any) {
+  let data: Row[] = [...state.data];
+  switch (action.type) {
+    case "add":
+      data.push(action.payload);
+      return {
+        ...state,
+        data,
+      };
+    case "update":
+      data = state.data.map((x: any) =>
+        x.number === action.payload.number ? action.payload : x
+      );
+      return { ...state, data };
+    case "delete":
+      debugger;
+      data = state.data.filter((x: any) => x !== action.payload);
+      return { ...state, data };
+    case "set":
+      data = action.payload;
+      return {
+        ...state,
+        data,
+      };
+  }
```

Comment:
More of the stylistic preference, than an issue. 
```suggestion
  switch (action.type) {
    case "add":
      return {
        ...state,
        data: [...state.data, action.payload],
        // or alternative
        // data: state.data.concat(action.payload)
      };
    case "update":
      return {
        ...state,
        data: state.data.map((x: any) =>
          x.number === action.payload.number ? action.payload : x
        )
      };
    case "delete":
      return { ...state, data: state.data.filter((x: any) => x !== action.payload) };
    case "set":
      return {
        ...state,
        data: action.payload,
      };
  }
```